components/ca-import.js

```diff
@@ -0,0 +1,727 @@
+                        data: this.data,
+                        contentType: 'application/json'
+                    })
+                    .then(data => {
+                        trace('CSV file converted and posted');
+                        this.onsave.call(this, data);
+                    });
+            }
+        }
+
+       /**
+        * Executes once data has been posted to the server
+        * @param {array} data, table data
+        * @returns {undefined} nothing.
+        */
+        onsave(data) {
+            trace('saved');
+        }
+
+       /**
+        * Executes when the user adds a file
+        * @returns {undefined} nothing.
+        */
+        onfileadded() {
+            trace('user added file containing data');
+        }
+
+       /**
+        * Renders the HTML view of the data if present
+        * @param {array} headers to use for table
+        * @param {array} data to use for rows
+        * @returns {undefined} nothing. Builds table to DOM
+        */
+        render(headers, data) {
+            // eslint-disable-next-line no-param-reassign
+            headers = headers || this.headers;
+            // eslint-disable-next-line no-param-reassign
+            data = data || this._data;
+
+            if (data) {
+                const table = this.querySelector('table');
+                const colGroup = this.querySelector('colgroup');
+                const thead = this.querySelector('thead');
+                const tbody = this.querySelector('tbody');
+
+                // remove previous values
+                clearElement(colGroup);
+                clearElement(thead);
+                clearElement(tbody);
+
+                // remove any errors on this form
+                this.form.removeAttribute('data-error');
+                const formError = document.querySelector('ca-form [data-error]');
+                if (formError) {
+                    formError.removeAttribute('data-error');
+                }
+
+                if (headers.length > 0) {
+
+                    const tr = createElement(null, 'tr');
+
+                    // insert table headers
+                    for (let i = 0, l = headers.length; i < l; i++) {
+
+                        const label = headers[i].replace('*', '');
+                        const key = toCamelCase(label.trim());
+
+                        // insert selection col/header cell
+                        createElement(colGroup, 'col', { 'data-key': key });
+
+                        // insert header cell into header row
+                        const th = createElement(tr, 'th', { 'data-key': key });
+
+                        // insert value within a span to allow us to alter its value via CSS (hide span, insert content before etc)
+                        createElement(th, 'span', null, label);
+                    }
+
+                    thead.appendChild(tr);
+                }
+
+                data.forEach((item, index) => {
```

Comment:
can't see it using index here, can it be removed?  why isn't linting moaning about that?